inc/vmem.h

```diff
@@ -0,0 +1,33 @@
+
+#include <iostream>
+#include <deque>
+#include <map>
+
+#define VMEM_RAND_FACTOR 91827349653
```

Comment:
I don't understand the utility of this. Why is this needed?

Reply:
It's a number that came from mashing my keyboard.  It's used in the algorithm just to make sure that we don't get stuck in a loop of the RNG always producing 0s.

I tested the distribution of numbers generated by this RNG in a separate program, and I found it to be acceptable.